client/transport.go

```diff
@@ -21,7 +21,9 @@ func (tf transportFunc) RoundTrip(req *http.Request) (*http.Response, error) {
 func resolveTLSConfig(transport http.RoundTripper) (*tls.Config, error) {
 	switch tr := transport.(type) {
 	case *http.Transport:
-		return tr.TLSClientConfig, nil
+		if tr.DialTLS != nil {
+			return tr.TLSClientConfig, nil
+		}
 	case transportFunc:
 		return nil, nil // detect this type for testing.
 	default:
```

Comment:
wait no, sorry, don't do as I said :) I tested and it's wrong - just add a `return nil, nil` after line 26